oktapam/resource_security_policy.go

```diff
@@ -0,0 +1,869 @@
+package oktapam
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/okta/terraform-provider-oktapam/oktapam/client"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/descriptions"
+)
+
+func resourceSecurityPolicy() *schema.Resource {
+	return &schema.Resource{
+		Description:   descriptions.ResourceSecurityPolicy,
+		CreateContext: resourceSecurityPolicyCreate,
+		ReadContext:   resourceSecurityPolicyRead,
+		DeleteContext: resourceSecurityPolicyDelete,
+		UpdateContext: resourceSecurityPolicyUpdate,
+		Schema: map[string]*schema.Schema{
+			attributes.ID: {
+				Type:     schema.TypeString,
+				Computed: true,
+				// Description is autogenerated
+			},
+			attributes.Name: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.Name,
+				ValidateDiagFunc: func(i any, p cty.Path) diag.Diagnostics {
```

Comment:
nit You can use the inbuilt validation functions too.

https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation

There is a method to convert legacy validate func - https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation#ToDiagFunc